app/models/annotations/dynamic.rb

```diff
@@ -121,14 +121,33 @@ def update_elasticsearch_dynamic
 
   def add_update_elasticsearch_dynamic(op)
     return if self.disable_es_callbacks
+    handle_elasticsearch_response(op)
     op = 'create_or_update' if annotation_type == 'smooch'
     options = get_elasticsearch_options_dynamic
     options.merge!({op: op, nested_key: 'dynamics'})
     add_update_nested_obj(options)
   end
 
+  def handle_elasticsearch_response(op)
+    if self.annotation_type =~ /^task_response/ && self.annotated_type == 'Task'
+      allowed_responses = %w(task_response_multiple_choice task_response_single_choice task_response_free_text)
+      if allowed_responses.include?(self.annotation_type)
```

Comment:
This condition could be in the first `if` already, instead of `self.annotation_type =~ /^task_response/`

Reply:
Done